torch/_inductor/config.py

```diff
@@ -361,6 +361,19 @@ def decide_compile_threads():
 # Pad input tensors of matmul/bmm/addmm to leverage Tensor Cores in NVIDIA GPUs
 shape_padding = os.environ.get("TORCHINDUCTOR_SHAPE_PADDING", "1") == "1"
 
+# When, during shape padding, dimension N would have to be padded, but
+# dimension M would not, then we can avoid a padding if we perform an
+# explicit transpose ( e.g. matmul(A,B) = matmul(B.T, A.T) ).T in order to
+# put the M dimension in the N position, therefore ensuring an aligned
+# GEMM result without padding. This can have dramatic
+# performance benefits if it is possible. Also, if this flag is enabled,
+# dimension M is not padded if N is aligned, since that's unneccessary
+# for an aligned result.
+shape_pad_use_transpose: bool = True
+
+# Whether to always use shape padding if it is enabled and possible
+shape_pad_always: bool = False
```

Comment:
nit: `force_shape_pad` is more in line with the rest of the flags, `force_layout_optimization` etc